apps/challenges/aws_utils.py

```diff
@@ -0,0 +1,542 @@
+        return {"Error": message, "ResponseMetadata": {"HTTPStatusCode": 400}}
+
+
+def update_service_by_challenge_pk(client, challenge, num_of_tasks, forceNewDeployment=False):
+    """
+    Updates the worker service for a challenge, and scales the number of workers to num_of_tasks.
+
+    Parameters:
+    client (boto3.client): the client used for making requests to ECS
+    challenge (<class 'challenges.models.Challenge'>): The challenge object  for whom the task definition is being registered.
+    num_of_tasks (int): Number of workers to scale to for the challenge.
+    forceNewDeployment (bool): Set True (mainly for restarting) to specify if you want to redploy with the latest image from ECR. Default is False.
+
+    Returns:
+    dict: The response returned by the update_service method from boto3. If unsuccesful, returns an error dictionary
+    """
+
+    queue_name = challenge.queue
+    service_name = "{}_service".format(queue_name)
+    task_def_arn = challenge.task_def_arn
+
+    kwargs = update_service_args.format(service_name=service_name, task_def_arn=task_def_arn, forceNewDeployment=forceNewDeployment)
+    kwargs = eval(kwargs)
+
+    try:
+        response = client.update_service(**kwargs)
+        if response["ResponseMetadata"]["HTTPStatusCode"] == HTTPStatus.OK:
+            challenge.workers = num_of_tasks
+            challenge.save()
+        return response
+    except ClientError as e:
+        return e.response
+
+
+def delete_service_by_challenge_pk(challenge):
+    """
+    Deletes the workers service of a challenge.
+
+    Before deleting, it scales down the number of workers in the service to 0, then proceeds to delete the service.
+
+    Parameters:
+    challenge (<class 'challenges.models.Challenge'>): The challenge object for whom the task definition is being registered.
+
+    Returns:
+    dict: The response returned by the delete_service method from boto3
+    """
+    client = get_boto3_client("ecs", aws_keys)
+    queue_name = challenge.queue
+    service_name = "{}_service".format(queue_name)
+    kwargs = delete_service_args.format(service_name=service_name, force=True)
+    kwargs = eval(kwargs)
+    try:
+        if(challenge.workers != 0):
+            response = update_service_by_challenge_pk(client, challenge, 0, False)
+            if (response["ResponseMetadata"]["HTTPStatusCode"] != HTTPStatus.OK):
+                return response
+
+        response = client.delete_service(**kwargs)
+        if response["ResponseMetadata"]["HTTPStatusCode"] == HTTPStatus.OK:
+            challenge.workers = None
+            challenge.save()
+            client.deregister_task_definition(taskDefinition=challenge.task_def_arn)
+            challenge.task_def_arn = ""
+            challenge.save()
+        return response
+    except ClientError as e:
+            return e.response
+
+
+def service_manager(client, challenge, num_of_tasks=None, forceNewDeployment=False):
```

Comment:
Let's try to use underscrore convention only here while listing arguments. Replace `forceNewDeployment` -> `force_new_deployment`